packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -623,6 +623,8 @@ abstract class ResidentRunner {
   }
 
   Future<Null> processTerminalInput(String command) async {
+    // When terminal doesn't support line mode, '\n' can sneak into the input.
+    command = command.replaceAll('\n', '');
```

Comment:
trim() and replaceAll() have different semantics. Which are you expecting? Do you just want to remove a trailing LF?